### ISSUE TITLE
[Bug]: Getting NPE for a simple test with the Dead Code Elimination

### ISSUE THREAD

**Issue by TharmiganK** [ballerina-platform/ballerina-lang#43552]
### Description

Getting the following error when running a simple test assertion with string `includes` API.

This is found while checking url module with dead code elimination(Related task - https://github.com/ballerina-platform/ballerina-library/issues/7339) 

```console
[2024-11-06 13:02:06,298] SEVERE {b7a.log.crash} - Cannot invoke "io.ballerina.runtime.internal.values.FPValue.getFunction()" because "<local1>" is null 
java.lang.NullPointerException: Cannot invoke "io.ballerina.runtime.internal.values.FPValue.getFunction()" because "<local1>" is null
        at tharmigan.package$test.0.tests.test.test(tests/test.bal:6)
        at tharmigan.package$test.0.lambdas.$_generated1tests.test_execute-generated_1.test$lambda0$(tests/test_execute-generated_1.bal:4)
        at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(SchedulerItem.java:54)
        at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:324)
        at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:291)
        at java.base/java.lang.Thread.run(Thread.java:840)
```

### Steps to Reproduce

Try to run the following test with dead code elimination feature:
```bal
import ballerina/test;

@test:Config
function test() {
    string result = "a";
    test:assertTrue(result.includes(" "));
}
```

```console
Compiling source
        tharmigan/package:0.1.0
Duration for unused BIR node analysis : 1ms
    tharmigan/package:0.1.0 : 1ms


Running Tests

        package
[2024-11-06 13:02:06,298] SEVERE {b7a.log.crash} - Cannot invoke "io.ballerina.runtime.internal.values.FPValue.getFunction()" because "<local1>" is null 
java.lang.NullPointerException: Cannot invoke "io.ballerina.runtime.internal.values.FPValue.getFunction()" because "<local1>" is null
        at tharmigan.package$test.0.tests.test.test(tests/test.bal:6)
        at tharmigan.package$test.0.lambdas.$_generated1tests.test_execute-generated_1.test$lambda0$(tests/test_execute-generated_1.bal:4)
        at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(SchedulerItem.java:54)
        at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:324)
        at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:291)
        at java.base/java.lang.Thread.run(Thread.java:840)
 

        test has failed.


                [fail] test:

                    error("java.lang.NullPointerException: Cannot invoke "io.ballerina.runtime.internal.values.FPValue.getFunction()" because "<local1>" is null")
                                callableName: test moduleName: tharmigan.package$test.0.tests.test fileName: tests/test.bal lineNumber: 6
                                callableName: test$lambda0$ moduleName: tharmigan.package$test.0.tests.test_execute-generated_1 fileName: tests/test_execute-generated_1.bal lineNumber: 4



                0 passing
                1 failing
                0 skipped

                Test execution time : 0.048s
error: there are test failures
```

### Affected Version(s)

Ballerina SwanLake Update 11(2201.11.x)

### Related area

-> Compilation

**Comment by MaryamZi** [comment]
Seems to be happening with langlib function calls.

Reproducible when the function is in the source too, rather than tests. 

Source

```ballerina
public function includesFn(string a, string b) returns boolean {
    return a.includes(b);
}
```

Tests

```ballerina
import ballerina/test;

@test:Config
function test() {
    boolean res = includesFn("foo", "bar");
    test:assertFalse(res);
}
```

Issue could be related to `io.ballerina.projects.UsedBIRNodeAnalyzer.InvocationData#registerNodes` since the function pool is not populated for langlibs.

**Comment by MaryamZi** [comment]
Seems like the error goes away when the start index is provided.

E.g.,

```ballerina
return a.includes(b, 4);
```

**Comment by MaryamZi** [comment]
We've added some fixes which are available with `2201.11.0-20241111-101700-af28a18f`, can you please check with this version?
